scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -314,8 +326,24 @@ def parse_generated(self):
         if self.platform.name == "unit_testing":
             return {}
 
-        cmake_cache_path = os.path.join(self.build_dir, "CMakeCache.txt")
-        defconfig_path = os.path.join(self.build_dir, "zephyr", ".config")
+        if self.testsuite.sysbuild:
+            # We must parse the domains.yaml file to determine the
+            # default sysbuild application
```

Comment:
do we really need to do that ?
The `self.source_dir` should be the default sysbuild application so any special reason that cannot be used here or that it is undesired to use it ?

I can envision future complex scenarios where we want to use sysbuild for multiple samples where default makes less sense, but I don't beleive that's the case here.

Reply:
We do I believe, because twister does not parse the source application, it parses the generated `.config` and `edt.pickle` files in the build folder for an application. This leaves us with the following options:

- Don't parse `domains.yaml`, and simply search the root of the build folder: This will result in searching the "meta" sysbuild application, that includes all other applications in the build as external projects. This will work, but we won't be able to do useful Kconfig or devicetree filtering
- Parse the `domains.yaml` and select the default application's build directory: This will enable Kconfig and devicetree filtering on the primary application only. It's the solution I have enabled currently due to the low complexity and impact
- Parse the `domains.yaml` and generate filter data for all applications: This is probably the best solution long term, but I avoided it in this PR because I wasn't certain how best to handle instructing twister which domain you wanted a filter to apply to

Reply:
> We do I believe, because twister does not parse the source application, it parses the generated .config and edt.pickle files in the build folder for an application. This leaves us with the following options:

Thanks for the explanation.
The path you have chosen makes most sense at this stage.